### PR TITLE
fix(chart/segment): check that query is valid before executing it

### PR DESCRIPTION
**Pull request by VincentMolinie**
### General

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Test manually the implemented changes
- [ ] Validate the code quality (indentation, syntax, style, simplicity, readability)

### Security

- [ ] Consider the security impact of the changes made

### CODE REVIEW

File: packages/forestadmin-client/package.json [hunk 1]
```diff
@@ -15,7 +15,8 @@
     "jsonwebtoken": "^8.5.1",
     "lru-cache": "^7.3.1",
     "object-hash": "^3.0.0",
-    "superagent": "^7.0.1"
+    "superagent": "^7.0.1",
+    "@forestadmin/datasource-toolkit": "1.0.1-alpha.1"
```
Comment (VincentMolinie) [inline packages/forestadmin-client/package.json:19]: @ghusse is that an issue to pass it as a dependencies ? I had to do that be cause I'm extending `UnprocessableError`

Comment (ghusse) [reply packages/forestadmin-client/package.json:19]: Yes, we only wanted to use the types and not require it for implementations.

File: packages/agent/test/services/authorization/authorization.test.ts [hunk 2]
```diff
@@ -1,6 +1,9 @@
 import { ChartType, CollectionActionEvent } from '@forestadmin/forestadmin-client';
 import { Collection } from '@forestadmin/datasource-toolkit';
 import { Context } from 'koa';
+import ChainedSQLQueryError from '@forestadmin/forestadmin-client/src/permissions/errors/chained-sql-query-error';
```
Comment (romain-gilliotte) [inline packages/agent/test/services/authorization/authorization.test.ts:4]: You should export those errors from the forstadmin/client package.

Comment (VincentMolinie) [reply packages/agent/test/services/authorization/authorization.test.ts:4]: Good catch, it's already exported I used a bad auto import 😅

**Review by adriguy — APPROVED** [review]

**Pull request merged**